fix(redshift): allow standard_conforming_strings option

aheuermann opened the pull request:
<!-- 
Thanks for wanting to fix something on Sequelize - we already love you!
Please fill in the template below.
If unsure about something, just do as best as you're able.

If your PR only contains changes to documentation, you may skip the template below.
-->

### Pull Request check-list

_Please make sure to review and check all of these items:_

- [x] Does `npm run test` or `npm run test-DIALECT` pass with this change (including linting)?
- [ ] Does the description below contain a link to an existing issue (Closes #[issue]) or a description of the issue you are solving?
- [x] Have you added new tests to prevent regressions?
- [x] Is a documentation update included (if this change modifies existing APIs, or introduces new ones)?
- [x] Did you follow the commit message conventions explained in [CONTRIBUTING.md](https://github.com/sequelize/sequelize/blob/master/CONTRIBUTING.md)?

<!-- NOTE: these things are not required to open a PR and can be done afterwards / while the PR is open. -->

### Description of change

We use sequelize with Redshift using the sequelize postgres dialect. It doesn't support the `standard_conforming_strings` option since it is based on an older version of postgres.  Since https://github.com/sequelize/sequelize/commit/850c7fd04669e0fef9238b6dc4f8d6ee93ed71e9 was introduced,  sequelize will try to set this option and fail, so I added a config option.

SimonSchick commented on types/lib/sequelize.d.ts:
```diff
@@ -305,6 +305,15 @@ export interface Options extends Logging {
+   * The PostgreSQL `standard_conforming_strings` session parameter. Set to `false` to not set the option.
+   * WARNING: Setting this to false may expose vulnerabilities and is not recommended!
+   *
+   * @default all aliases
```
Nit: Should be `true`.

SimonSchick commented on types/lib/sequelize.d.ts:
```diff
@@ -305,6 +305,15 @@ export interface Options extends Logging {
    */
   operatorsAliases?: OperatorsAliases | false;
```
Optional: While we are here, can you remove `| false` here pretty please? :)

aheuermann commented:
Fixed, thanks for taking a look!

I removed the tests as they weren't really testing anything since the  default is `on` and the query to set the option runs on connect and is per-session. I couldn't find a good way to spy on the query itself. If you have any suggestions let me know, and I can add a test.

sushantdhiman commented:
:tada: This PR is included in version 5.7.4 :tada:

The release is available on:
- [npm package (@latest dist-tag)](https://www.npmjs.com/package/sequelize)
- [GitHub release](https://github.com/sequelize/sequelize/releases/tag/v5.7.4)

Your **[semantic-release](https://github.com/semantic-release/semantic-release)** bot :package::rocket:

sushantdhiman commented:
This should be support by different `dialect = redshift` or version check, it can leave users open to SQL injection if redshift updated their postgres version
